packages/language-server/src/xml-view-diagnostics.ts

```diff
@@ -26,24 +27,30 @@ export function getXMLViewDiagnostics(opts: {
     xmlView: xmlDocAst,
     model: opts.ui5Model,
   });
-  const diagnostics = validationIssuesToLspDiagnostics(issues, opts.document);
+  const diagnostics = validationIssuesToLspDiagnostics({
+    issues,
+    document: opts.document,
+    ui5Model: opts.ui5Model,
+  });
   return diagnostics;
 }
 
-function validationIssuesToLspDiagnostics(
-  issues: UI5XMLViewIssue[],
-  document: TextDocument
-): Diagnostic[] {
-  const diagnostics: Diagnostic[] = map(issues, (currIssue) => {
+function validationIssuesToLspDiagnostics(opts: {
+  issues: UI5XMLViewIssue[];
+  document: TextDocument;
+  ui5Model: UI5SemanticModel;
+}): Diagnostic[] {
+  const diagnostics: Diagnostic[] = map(opts.issues, (currIssue) => {
     const commonDiagnosticPros: Diagnostic = {
       range: {
-        start: document.positionAt(currIssue.offsetRange.start),
+        start: opts.document.positionAt(currIssue.offsetRange.start),
         // Chevrotain's end offsets are none inclusive
-        end: document.positionAt(currIssue.offsetRange.end + 1),
+        end: opts.document.positionAt(currIssue.offsetRange.end + 1),
       },
       severity: toLspSeverity(currIssue.severity),
       source: "UI5 Language Assistant",
-      message: currIssue.message,
+      message: convertDescriptionToMarkup(currIssue.message, opts.ui5Model)
```

Comment:
If the markdown isn't displayed in the problems view, why convert it to markdown? Wouldn't it be better to remove the formatting (which would require a new function)?

Reply:
this is a left over mistake.

Converting is also a problem when the message is huge.